### DISCUSSION TITLE
Share data between multiple RQ Clients in one page

### DISCUSSION THREAD

**Discussion by ismoiliy98** [TanStack/query#3472] [Q&A] [answered]
Hello. I am building a react library that will be bundled and can be used as a library (widget) on a page. Let's imagine a weather widget. The widget can be included in multiple places on the page. Does RQ share data between multiple widgets? e.g. Widget #1 already fetched data, will RQ Client in widget #2 reuse that fetched data or vice-versa?

**Comment by lm-chad-simmons** [comment]
If both widgets are rendered as a child of the same QueryClientProvider they will share the same QueryCache

**Reply by ismoiliy98** [reply]
No, they are using the same bundle but loaded inside different DOM elements with separate react instances. Chunks are shared between each widget.

**Comment by gareth-canva** [comment, marked as answer]
Would the `contextSharing` prop on `QueryClientProvider` do what you want? [Docs](https://react-query.tanstack.com/reference/QueryClientProvider)

```
contextSharing: boolean
defaults to false

Set this to true to enable context sharing, which will share the first and at least one instance of the context
across the window to ensure that if React Query is used across different bundles or microfrontends
they will all use the same instance of context, regardless of module scoping.
```

**Reply by ismoiliy98** [reply]
Exactly! Thank you very much
